core/src/main/java/org/elasticsearch/common/util/BigArrays.java

```diff
@@ -435,15 +440,18 @@ public CircuitBreakerService breakerService() {
 
     private <T extends AbstractBigArray> T resizeInPlace(T array, long newSize) {
         final long oldMemSize = array.ramBytesUsed();
+        assert oldMemSize == array.ramBytesEstimated(array.size) :
+            "ram bytes used should equal that which was previously estimated";
```

Comment:
Can you add the `oldMemSize` and new estimate to the assertion message? (In case it fails)